tests/conftest.py

```diff
@@ -197,3 +218,39 @@ def cylc_validate_cli(capsys, caplog):
 @pytest.fixture(scope='module')
 def mod_cylc_validate_cli(mod_capsys, mod_caplog):
     return _cylc_validate_cli(mod_capsys, mod_caplog)
+
+
+@pytest.fixture
+def file_poll():
+    """Poll for the existance of a file.
+    """
+    def _inner(fpath: "Path", timeout: int = 5, file_exists=True):
+        for _ in range(timeout):
+            if file_exists and fpath.exists():
+                break
+            elif not file_exists and not fpath.exists():
+                break
+            sleep(1)
+        else:
+            raise TimeoutError(
+                f'file {fpath} not found after {timeout} seconds')
+    return _inner
+
+
+@pytest.fixture
+def cylc_stop(file_poll):
+    """Cylc stop & check the contact file"""
+    def _inner(id_):
+        run(split(f'cylc stop --now --now {id_}'))
+        contactfile = Path.home() / f'cylc-run{id_}.service/contact'
```

Comment:
Shouldn't this be?
```suggestion
        contactfile = Path.home() / f'cylc-run/{id_}/.service/contact'
```
Or better yet
```python
from cylc.flow.workflow_files import get_contact_file_path
```